tests/conftest.py

```diff
@@ -602,3 +604,29 @@ def create_mock_http_server():
     mock_server_thread.start()
 
     return mock_server_port
+
+
+def pytest_ignore_collect(path):
```

Comment:
Ah, I guess it's because of the type hints in the integration that we import? That would be another argument for turning them into old-style type hints.

Reply:
> What makes the gql tests incompatible with older python versions? I see some f-strings which would be easy to make compatible, is there anything else? I'd honestly prefer to write the tests in a way that all supported py versions can load them unless we have some issue that's very hard to fix.

It looks like just the f-strings and a few uses of the `*`-unpacking operator are causing the Syntax Error. So, we can probably fix this if you would like us to be able to load the test in all supported Python versions. 

Although I am just wondering what your reasoning is for why we should write this test so it can be loaded in Python versions below 3.7? GQL is only compatible with Python 3.7+, and we are anyways skipping all the tests in Python versions below 3.7, so to me it does not seem like we lose anything by skipping loading the file in older Python versions. I also think we could use these sorts of minimum version suffixes elsewhere in our test suite. 


> Ah, I guess it's because of the type hints in the integration that we import? That would be another argument for turning them into old-style type hints.

The type hints in the integration appear to be fine. I believe this is because we import the integration only after attempting to load the gql library with `pytest.importorskip("gql")`, and so PyTest will never attempt to load the integration in Python versions that don't support type hints (since all versions that gql supports also support type hinting).

Reply:
My concern is mostly that we're changing the test loading mechanism itself to do something that seems relatively easy to achieve by writing the test in a slightly different way. I'm also wondering if the unsupported syntax in the test module could create issues somewhere outside of the context of a test runner, maybe for packaging? (Don't know if anyone is packaging the SDK for py2, though, and if so, if the tests are involved in the process in any way.) It seems safer to just make the test compatible.

Reply:
Fair enough, made the change